cms_form/static/src/js/ajax.js

```diff
@@ -0,0 +1,73 @@
+odoo.define('cms_form.ajax', function (require) {
+    'use strict';
+
+    var core = require('web.core'),
+        animation = require('website.content.snippets.animation');
+
+    animation.registry.CMSFormAjax = animation.Class.extend({
+        selector: ".cms_form_wrapper form[data-ajax]",
+        events: {
+            'submit': 'submit_form',
+        },
+        start: function () {
+            this.data = this.$el.data('form');
+            if (this.$el.data('ajax-onchange')) {
+                this.$el.on('change', this.proxy('submit_form'));
+            }
+            this.$el.parents('.cms_form_wrapper').find(
+                this.data.form_content_selector
+            ).find(
+                '.pagination a[href]'
+            ).on(
+                'click', this.proxy('pager')
+            );
+        },
+        ajax_submit: function (additional_data) {
+            return jQuery.ajax(
+                _.str.sprintf('/cms/ajax/search/%s', this.data.model),
+                {
+                    data:
+                    this.$el.serialize() + '&csrf_token=' + core.csrf_token +
+                    (additional_data || ''),
+                    dataType: 'json',
+                    success: this.proxy('success'),
+                    error: this.proxy('error'),
+                }
+            );
+        },
+        submit_form: function (ev) {
+            var $container = this.$container();
+
+            jQuery.blockUI();
```

Comment:
I  tested it on runbot: looks good but UI  blocking and spinner are a bit ugly.
Can we delegate this to internal function to ease override? Eg: `this.blockUI()` +  `this.unBlockUI()`.
You can add just a TODO  or an item on the roadmap if you don't have time :wink:

Reply:
I did the roadmap entry